DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/TokenCheckMiddleware.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+
+using GitHubExtension.Infrastructure.Constants;
+using GitHubExtension.Infrastructure.Extensions.Owin;
+
+using Microsoft.Owin;
+
+namespace GitHubExtension.EntryPoint
+{
+    /// <summary>
+    /// Middleware that check external access token presence in claims
+    /// </summary>
+    public class TokenCheckMiddleware : OwinMiddleware
+    {
+        private const string NoToken = "You don't have access token to enter this resource";
+
+        public TokenCheckMiddleware(OwinMiddleware next)
+            : base(next)
+        {
+        }
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            if (context.IsAuthenticated())
+            {
+                if (!context.Authentication.User.HasClaim(c => c.Type == ClaimConstants.ExternalAccessToken))
+                {
+                    context.Authentication.SignOut();
```

Comment:
As you can see it's nice place to verify existing claim due to you have all necessesary access to all you need methods in one place. Btw do you think how we ca test our middleware?